scripts/GenerateConfig.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/python3
+
+import configparser
+import sys
+import getopt
+import os
+import time
+from pathlib import Path
+
+helpMessage = """
+Usage:
+    GenerateConfig.py --input /path/to/fasta
+        [--input /path/to/fasta-2 ...]
+
+Note: The goal of this script is to generate a good starter configuration for the
+Shasta long read assembler. 
+
+Each genome is different and reads differ in quality. So it is impossible to generate a perfect
+configuration programmatically. If you'd like to get some suggestions on how to tweak the
+configuration generated by this script, you should use the `GetFeedback.py` script after 
+you run your assembly.
+
+If you're still not happy with the assembly after using these two scripts, please file a Github
+issue and we can help.
+"""
+
+def usage():
+    print(helpMessage)
+    return
+
+def getNextRead(inputFile):
+    fileType = Path(inputFile).suffix
+    if (fileType not in ['.fasta', '.fastq']):
+        raise Exception('Input files need to be in `fasta` or `fastq` format')
+    
+    isFastq = fileType == '.fastq'
+
+    isNextLineAReadInFastq = False
+    with open(inputFile) as fp:
+        for line in fp:
+            line = line.strip()
+            if isFastq:
+                if isNextLineAReadInFastq:
+                    yield line
+                    isNextLineAReadInFastq = False
+                elif line[0] == '@':
+                    isNextLineAReadInFastq = True
+                else:
+                    isNextLineAReadInFastq = False
+            else:
+                if line[0] != '>':
+                    yield line
+                
+
+
+def getMinReadLength(inputFiles, genomeSize, baseCallerId):
+    # Reads map to a bin that is int(read-length / 1000).
+    binnedHistogramOfBases = [0] * 1000
+    startTime = time.time()
+
+    for inputFile in inputFiles:
```

Comment:
Instead of reading the entire file, I can read a smaller fraction of it (say 10%) and compute my stats based on that. I plan to do that in a separate PR shortly.